gamemode/modules/base/cl_jobmodels.lua

```diff
@@ -42,11 +44,22 @@ local function sendModels() -- run after the jobs have loaded
     net.SendToServer()
 end
 
-do
-    local models = sql.Query([[SELECT jobcmd, model FROM darkp_playermodels;]])
+local function jobHasModel(job, model)
+    return istable(job.model) and table.HasValue(job.model, model) or job.model == model
+end
+
+timer.Simple(0, function()
+    -- run after the jobs have loaded
+    local models = sql.Query([[SELECT jobcmd, model, server FROM darkp_playermodels;]])
+
     for _, v in ipairs(models or {}) do
+        if v.server and v.server ~= game.GetIPAddress() then continue end
```

Comment:
This could be made somewhat more efficient by placing the server check in the query rather than in the code. That way, there is less data sent between SQLite and Lua, and you don't need to walk over as much.

Could you change the select query to be as follows?

```sql
    local models = sql.Query(string.format([[SELECT jobcmd, model FROM darkp_playermodels WHERE server = %s;]], sql.SQLStr(game.GetIPAddress())))
```

Reply:
better todo it on lua side, because it will use old rows without `server` column.  
if do this on the sql side, as you suggested, then we should add clearing table of old data without the server column, because they will be useless.

Reply:
Good point! Maybe it's still possible to place the condition in the query by using `WHERE server IS NULL OR server = %s`?

Reply:
lgtm